java/BUILD.bazel

```diff
@@ -0,0 +1,235 @@
+load("@com_github_checkstyle_java//checkstyle:checkstyle.bzl", "checkstyle_test")
+load("//bazel:ray.bzl", "flatbuffer_java_library")
+
+exports_files([
+    "testng.xml",
+    "checkstyle.xml",
+    "checkstyle-suppressions.xml",
+])
+
+java_binary(
+    name = "bazel_deps",
+    main_class = "com.github.johnynek.bazel_deps.ParseProject",
+    runtime_deps = ["@bazel_deps//jar"],
+)
+
+java_import(
+    name = "org_ray_ray_java_root",
+    jars = [
+        "liborg_ray_ray_api.jar",
+        "liborg_ray_ray_api-src.jar",
+        "liborg_ray_ray_runtime.jar",
+        "liborg_ray_ray_runtime-src.jar",
+        "liborg_ray_ray_runtime_test.jar",
+        "liborg_ray_ray_runtime_test-src.jar",
+        "liborg_ray_ray_test.jar",
+        "liborg_ray_ray_test-src.jar",
+        "org_ray_ray_tutorial_deploy.jar",
+        "org_ray_ray_tutorial_deploy-src.jar",
+        "all_tests_deploy.jar",
+        "all_tests_deploy-src.jar",
+    ],
+    deps = [
+        ":org_ray_ray_api",
+        ":org_ray_ray_runtime",
+        ":org_ray_ray_tutorial",
+        ":org_ray_ray_test",
+        ":org_ray_ray_runtime_test",
+    ],
+)
+
+java_library(
+    name = "org_ray_ray_api",
```

Comment:
there's some duplicated code in this file that defines `java_library`, and `checkstyle_test` for each module. I think we can define a function to consolidate the code. E.g.,
```
def define_java_module(name, additional_resources, define_test_lib = False,  **kwargs):
    java_library(
        name = "org_ray_ray_" + name,
        src = glob([name + "/src/main/java/**/*.java"]),
        resources = glob(...) + additional_resources,
        **kwargs,
     )
     checkstyle_test(...)
     if define_test_lib:
         java_libarary(...)
```